### DISCUSSION TITLE
v2 vue组件 getGraph

### DISCUSSION THREAD

**Discussion by kanocence** [antvis/X6#3069] [Q&A] [answered]
v2版本移除了 getGraph ，有没有替代的办法。
我需要 graph.trigger 触发自定义事件。

`export default {
  inject: ['getGraph', 'getNode'],
}`
> [Vue warn]: Injection "getGraph" not found

**Comment by lloydzhou** [comment]
你可以看看getNode拿到的节点是不是也能从某个属性拿到graph?

**Reply by NewByVector** [reply]
在 vue 组件中通过 provide 透出 graph 实例，这样会有什么问题吗？如果没有问题的话，我打算这么做。

**Reply by lloydzhou** [reply]
1. 如果使用了vue3开启teleport，那么按照正确的方式组织整个结构的时候是可以的。
2. 如果不满足上面的条件，那么可能就拿不到graph...
3. 另外还要考虑到如果开启minimap的时候传递graph的问题

**Comment by NewByVector** [comment, marked as answer]
在这个 [PR](https://github.com/antvis/X6/pull/3072) 中已经修复。

**Reply by lloydzhou** [reply]
有一个疑问：
1. 为啥react-shape支持通过props传递node和graph而vue-shape不支持
2. 而且中间的1.x有一个版本，我是加上了这个支持，后来迁移2.x的时候又移除了

**Comment by kanocence** [comment]
我使用的vue2.7，更新 @antv/x6-vue-shape 2.0.8 后
this.getGraph();  // undefined
this.getNode(); // 正常

**Reply by NewByVector** [reply]
确实有点问题，修复中 [PR](https://github.com/antvis/X6/pull/3078)
